.github/workflows/pypi.yml

```diff
@@ -74,14 +65,12 @@ jobs:
         uses: pypa/cibuildwheel@v2.22.0
         with:
           output-dir: dist
-        env:
-          CIBW_ARCHS_MACOS: "${{ matrix.macos_arch }}"
 
       - name: Upload wheels as workflow artifacts
         uses: actions/upload-artifact@v4
         with:
-          name: ${{ matrix.name }}-mypyc-wheels
+          name: ${{ runner.os }}-${{ runner.arch }}-mypyc-wheels
           path: ./dist/*.whl
 
-      - name: 🚢 Publish to PyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
+      # - name: 🚢 Publish to PyPI
+      #   uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
why did you upload here instead of have separate steep to collect all wheels anbd upload only if all platfrom buildt successfully?

Reply:
But you may then upload to pypi broken release that was revealed during build wheel.

Reply:
well, I'd call it a "partial" release.  it is actually pretty easy to upload more wheels if only some of them fail to build.  In any case, this whole thing is draft, and I'm actively working on it.  Nothing you see here at the moment should be considered something I've thought hard about.  please hang on a moment.  I'll let you know when it's ready for review if you'd like to look

Reply:
ok, short of uncommenting out the pypi publish step, i think this is now better.  let me know if it's following all the best practices you would suggest